triton-client/src/components/DatasetCard.tsx

```diff
@@ -23,13 +24,14 @@ export interface StagingAction {
 function DatasetCard({ datasetID }: DatasetCardProps) {
 	const dispatch = useAppDispatch()
 	const dataset = useAppSelector((state) => state.datasetsState.datasetsById[datasetID])
+	const requests = useAppSelector((state) => selectRequestByDatasetId(state, datasetID))
+
+	const readsetsById = useAppSelector((state) => state.readsetsState.readsetsById)
 	const project = useAppSelector((state) => dataset?.external_project_id ? state.projectsState.projectsById[dataset.external_project_id] : undefined)
 	const constants = useAppSelector(selectConstants)
-	const readsetsByDatasetId = useAppSelector((state) => state.readsetsState.readsetsByDatasetId)
-	const readsetsById = useAppSelector((state) => state.readsetsState.readsetsById)
-  const activeRequest = useMemo<DownloadRequest | undefined>(() => {
-    return dataset?.requests[0] ?? undefined
-  }, [dataset?.requests])
+	const activeRequest = useMemo<DownloadRequest | undefined>(() => {
+		return requests && requests[0] ? requests[0] : undefined
```

Comment:
Does request need to be represented by an array ? There is only one request by DatasetId to be expected.

Reply:
Oh right. Forgot about that.